charts/incubator/radicale/templates/_secret.tpl

```diff
@@ -0,0 +1,16 @@
+{{/* Radicale htpasswd file */}}
+{{- define "radicale.secret" -}}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: radicale-secret
+  labels:
+    {{- include "tc.common.labels" . | nindent 4 }}
+stringData:
+  users: |-
+    {{- range .Values.radicale.auth.users }}
+    {{ htpasswd .username .password }}
```

Comment:
I'm not familiar with this tpl syntax. 
What htpasswd represents here? The filename?

Any source for that syntax? I'm interested to know more :P

Reply:
It's a helm function: https://helm.sh/docs/chart_template_guide/function_list/#htpasswd

It renders the following:

```yaml
users:
  - username: hello
    password: world
  - username: foo
    password: bar
```

to this:

````text
hello:$2a$10$3KYHAwn7fL3uJknSLCMSDexyw88TK7dPKiE9zcuJW3rIVjpqYI3qO
foo:$2a$10$sAw97DZcgRCyd6jEAy4fo.0L.wTo.EPQcACxRGGG9ohZhi.fLvsvS
```

The function uses `bcrypt` to generate password hashes.

Reply:
Ahh I see thanks!